lib/xml/reader.dart

```diff
@@ -141,3 +144,147 @@ class XmlReader {
     return context.success(null, context.position + 1);
   }
 }
+
+/// A push based XmlReader interface, intended to be similar to .NET's XmlReader.
+class XmlTextReader {
+  /// Creates a new reader for `input`.
+  /// 
+  /// Setting `ignoreWhitespace` to false will cause text nodes 
+  XmlTextReader(String input, {this.ignoreWhitespace = true}) {
+    _result = Success(input, 0, null);
+    _depth = 0;
+    _eof = false;
+  }
+
+  /// If true, will ignore `XmlNodeType.TEXT` and when it is composed of whitespace.
+  bool ignoreWhitespace;
+
+  /// Parsing context.
+  Result _result;
+
+  /// The [XmlNodeType] of the current position of the reader.
+  XmlNodeType get nodeType => _nodeType;
```

Comment:
I suspect this will be confusing to write a switch statement over this, because not all node types are ever returned.

Reply:
switch statement is exactly how this intended to be used.  You call `read`, for example in a `while` loop, and check what kind of nodeType you have.

It doesn't use all of them - perhaps it would be worth having a different enum, like `XmlReaderNodeType` or something to remove the ones that would never be used (or expand this to have `read` start with `DOCUMENT` or somesuch).